### DISCUSSION TITLE
How to mock a module with bun:test?

### DISCUSSION THREAD

**Discussion by raulfdm** [oven-sh/bun#6236] [Q&A]
Hey folks.

I was considering migrating our company's unit test to use bun to gain some performance but then I didn't find a way of mocking like `vi.mock('module-name', ()=>{...}`.

I thought about importing the module and using `spyOn` but this would drastically change my current setup.

Am I missing something?

**Comment by roryokane** [comment]
It looks like Bun doesn’t support `jest.mock` or `vi.mock` yet. See open issue https://github.com/oven-sh/bun/issues/5394.

**Comment by KhalfaniW** [comment]
They have implented it in new bun versions according to the ticket https://github.com/oven-sh/bun/issues/5394
``` js
import { expect, test, mock } from "bun:test";
import { main, main2 } from "./module";

test("before", async () => {
  expect(main()).toBe(1);
  expect(main2()).toBe(2);
});

test("after", async () => {
  const mockDependency = () => "mocked";
  mock.module("./module.ts", () => ({ main: mockDependency }));
  expect(main()).toBe("mocked");
  expect(main2()).toBe(2);
});
```

**Reply by constraintAutomaton** [reply]
I wonder how can we revert the mock to the original implementation?

**Reply by brenodanyel** [reply]
Mocks are automatically reset between tests.
https://bun.sh/docs/cli/test#mocks
